crates/db_schema/src/lib.rs

```diff
@@ -277,12 +277,27 @@ pub enum CommunityVisibility {
   /// Public community, any local or federated user can interact.
   #[default]
   Public,
-  /// Unfederated community, only local users can interact.
-  LocalOnly,
+  /// Posts from this community do not appear on Local and All feeds.
+  Hidden,
+  /// Unfederated community, only local users can interact (with or without login).
+  LocalOnlyPublic,
+  /// Unfederated  community, only logged-in local users can interact.
+  LocalOnlyPrivate,
   /// Users need to be approved by mods before they are able to browse or post.
   Private,
 }
 
+impl CommunityVisibility {
+  pub fn can_federate(&self) -> bool {
+    use CommunityVisibility::*;
+    self != &LocalOnlyPublic && self != &LocalOnlyPrivate
+  }
+  pub fn can_view_without_login(&self) -> bool {
+    use CommunityVisibility::*;
+    self == &Public || self == &LocalOnlyPublic
+  }
+}
+
```

Comment:
It'd be nice if you could use these in the apub code that uses `match`, but I can't think of a way.